### PR TITLE
HW0

### PR DESCRIPTION
**Pull request by paulspam**
Сделал задачу 0 через циклы

### CODE REVIEW

File: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java [hunk 1]
```diff
@@ -23,5 +23,44 @@ public static void main(String[] args) {
                 new UserMeal(LocalDateTime.of(2015, Month.MAY, 31, 13, 0), "Обед", 500),
                 new UserMeal(LocalDateTime.of(2015, Month.MAY, 31, 20, 0), "Ужин", 510)
         );
+
+        getFilteredWithExceeded(mealList, LocalTime.of(7, 0), LocalTime.of(12, 0), 2000);
+//        .toLocalDate();
```
Comment (AndrewChertok) [inline src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java]: toLocalDate and toLokalTime releted to the next part of HW so it should't be here

File: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java [hunk 2]
```diff
@@ -23,5 +23,44 @@ public static void main(String[] args) {
...
+//        .toLocalTime();
+
+    }
+
+    public static List<UserMealWithExceed> getFilteredWithExceeded(List<UserMeal> mealList, LocalTime startTime, LocalTime endTime, int caloriesPerDay) {
+        //System.out.println("Need to return List");
+        Map<LocalDate, Integer> mapCaloriesPerDay = new HashMap<>();
+        for (UserMeal meal : mealList
+        ) {
```
Comment (AndrewChertok) [inline src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java]: Use for each in the following format: for(Object x : listObject) {
}

File: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java [hunk 3]
```diff
@@ -23,5 +23,44 @@ public static void main(String[] args) {
...
+            LocalDate date = meal.getDateTime().toLocalDate();
+            Integer newCalories = meal.getCalories();
+            mapCaloriesPerDay.merge(date, newCalories, Integer::sum);
+        }
+
+        List<UserMealWithExceed> mealWithExceedList = new ArrayList<>();
+        for (UserMeal meal : mealList
```
Comment (AndrewChertok) [inline src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java]: The same as above

File: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java [hunk 4]
```diff
@@ -23,5 +23,44 @@ public static void main(String[] args) {
...
+        ) {
+            if (TimeUtil.isBetween(meal.getDateTime().toLocalTime(), startTime, endTime)) {
+                if (mapCaloriesPerDay.get(meal.getDateTime().toLocalDate()) > caloriesPerDay) {
+                    mealWithExceedList.add(new UserMealWithExceed(meal.getDateTime(), meal.getDescription(),
+                            meal.getCalories(), true));
```
Comment (AndrewChertok) [inline src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java]: Use the following format when you need to carry the parameters: 
mealWithExceedList.add(
new UserMealWithExceed(meal.getDateTime(), meal.getDescription(),meal.getCalories(), true)
);

Comment (paulspam) [reply src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java]: Should I use tab in this way?
mealWithExceedList.add(
       new UserMealWithExceed(meal.getDateTime(), meal.getDescription(), meal.getCalories(), true));

**Comment by paulspam** [comment]
I correct all comments

File: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java [hunk 5]
```diff
@@ -23,5 +23,41 @@ public static void main(String[] args) {
...
                 new UserMeal(LocalDateTime.of(2015, Month.MAY, 31, 20, 0), "Ужин", 510)
         );
+
+        getFilteredWithExceeded(mealList, LocalTime.of(7, 0), LocalTime.of(12, 0), 2000);
+
+
+    }
+
+    public static List<UserMealWithExceed> getFilteredWithExceeded(List<UserMeal> mealList, LocalTime startTime, LocalTime endTime, int caloriesPerDay) {
+        //System.out.println("Need to return List");
```
Comment (AndrewChertok) [inline src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java:33]: Don't leave the code which you use for testing or commented useless text

**Pull request merged**